earthfile2llb/state_wait_item.go

```diff
@@ -0,0 +1,26 @@
+package earthfile2llb
+
+import (
+	"github.com/earthly/earthly/util/llbutil/pllb"
+	"github.com/earthly/earthly/util/waitutil"
+)
+
+type stateWaitItem struct {
```

Comment:
What is a `stateWaitItem`?

EDIT: Oh I see - it's a state we wait on to finish forced execution within a WAIT block, right?

Reply:
yeah it's quite tied to the converter, it was moved out of `wait_block.go` which was getting rather large,

`wait_block.go` contains some code that uses the stateWaitItem's converter reference, such as:

```
return item.c.forceExecution(ctx, *item.state, item.c.platr)
```

(where `c` is the `Converter`)

the higher-level methods of the wait items are defined as an interface under `util/waitutil`, but we need to keep all converter-references out of that to prevent cyclical dependencies.